File: cuckoo-hibersap-example-datamodel/src/main/java/de/akquinet/jbosscc/cuckoo/example/model/CustomerWithId.java
package de.akquinet.jbosscc.cuckoo.example.model;

import org.hibersap.annotations.BapiStructure;
import org.hibersap.annotations.Parameter;

@BapiStructure
public class CustomerWithId extends Customer
{
    @Parameter( "CUSTOMERID" )
    private String id;

    public String getId()
    {
        return id;
    }

    public void setId( String id )
    {
        this.id = id;
    }

    @Override
    public String toString()
    {
        return "CustomerWithId{" +
                "id='" + id + '\'' +
                "} " + super.toString();
    }
}

File: src/main/kotlin/org/hibersap/customer/model/CustomerWithId.kt
package org.hibersap.customer.model

import org.hibersap.annotations.BapiStructure
import org.hibersap.annotations.Parameter
import javax.validation.constraints.NotNull
import javax.validation.constraints.Size

@BapiStructure
data class CustomerWithId(
        @Parameter("CUSTOMERID") val id: String = "",
        @Parameter("CUSTNAME") var name: String = "",
        @Parameter("FORM") var formOfAddress: String = "",
        @Parameter("STREET") var street: String = "",
        @Parameter("POBOX") var poBox: String = "",
        @Parameter("POSTCODE") var postalCode: String = "",
        @Parameter("CITY") var city: String = "",
        @Parameter("COUNTR_ISO") @field:NotNull @field:Size(min = 2, max = 2) var countryKeyIso: String = "",
        @Parameter("REGION") var region: String = "",
        @Parameter("PHONE") var phoneNumber: String = "",
        @Parameter("EMAIL") var email: String = ""
)